Classes/Http/Middleware/Preview.php

```diff
@@ -73,24 +73,23 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
         if (!$config->validForSiteAndLanguage($site, $language)) {
             return $handler->handle($request);
         }
+
         // Store config in request
         $request = $request->withAttribute(self::REQUEST_ATTRIBUTE, $config);
 
-        $this->initializePreviewUser($language);
+        $this->initializePreviewUser($language, $site);
         $response = $handler->handle($request);
 
         // If the GET parameter PreviewUriBuilder::PARAMETER_NAME is set, then a cookie is set for the next request
         if ($request->getQueryParams()[PreviewUriBuilder::PARAMETER_NAME] ?? false) {
             /** @var NormalizedParams $normalizedParams */
             $normalizedParams = $request->getAttribute('normalizedParams');
             $cookie = new Cookie(
-                PreviewUriBuilder::PARAMETER_NAME,
-                $hash,
-                0,
-                $normalizedParams->getSitePath(),
-                '',
-                true,
-                true
+                name: PreviewUriBuilder::PARAMETER_NAME,
```

Comment:
sorry, this is PHP 8.x feature, but the current code supports also PHP 7.4

Reply:
not anymore we dont 😀